### PR TITLE
Feature: Ghost error & error pcts (#2)

### PR DESCRIPTION
**Pull request by elboy3**
https://app.shortcut.com/galileo/story/5478/dq-add-ghost-errors-to-pred-polygons

https://app.shortcut.com/galileo/story/5481/dq-add-error-pct-to-polygons-so-that-we-can-threshold-errors-on-the-api-side

### CODE REVIEW

File: dataquality/utils/semantic_segmentation/errors.py [hunk 1]
```diff
@@ -61,12 +35,40 @@ def calculate_misclassified_class(
...
+
+    :param preds: argmax of the prediction probabilities
+        shape = (height, width)
+    :param gold_masks: ground truth masks
+        shape =  height, width)
+    :param correct_class: the correct class of the polygon
+
+    returns: pixel accuracy of the predictions
+    """
+    relevant_region = gold_mask != BACKGROUND_CLASS
```
Comment (elboy3) [inline dataquality/utils/semantic_segmentation/errors.py:57]: just moving this fn a little lower

**Review by Ben-Epstein — APPROVED** [review]
approving with the context of a follow-up pr to rename ghost -> background and undetected -> missed"

File: dataquality/loggers/model_logger/semantic_segmentation.py [hunk 2]
```diff
@@ -146,6 +148,7 @@ def get_polygon_data(
                 golds.append(polygon.label_idx)
                 data_error_potentials.append(polygon.data_error_potential)
                 errors.append(polygon.error_type.value)
+                error_pcts.append(polygon.error_pct)
```
Comment (setu4993) [inline dataquality/loggers/model_logger/semantic_segmentation.py:151]: Just to confirm: gold won't have any overlap with preds, right? Else we'll see duplicates for the same polygon.

File: dataquality/schemas/semantic_segmentation.py [hunk 3]
```diff
@@ -18,6 +18,7 @@ class SemSegCols(str, Enum):
 class ErrorType(str, Enum):
     classification = "classification"
     undetected = "undetected"
+    ghost = "ghost"
```
Comment (setu4993) [inline dataquality/schemas/semantic_segmentation.py:21]: Is this how these are errors are referred to in research?

Comment (elboy3) [reply dataquality/schemas/semantic_segmentation.py:21]: Good question. I was wondering the same - check the #cv-taskforce slack channel i just started a discussion there! 

Also - in a follow up PR i am renaming this to "background" and renaming "undetected" to "missed" for consistency with OD

File: dataquality/loggers/model_logger/semantic_segmentation.py (hunk 2 above)
Comment (elboy3) [reply dataquality/loggers/model_logger/semantic_segmentation.py:151]: confirmed

**Comment by elboy3** [comment]
> Left a couple questions to improve my understanding, rest looks good.
> 
> Are there tests we can add here?

In DQ We are testing live for this sprint to get a run in sandbox 😅 

It's not being used by customers and isolated code, so for now no risk.

And then dedicating time after for robust testing

**Pull request merged**